pymodbus/client/base.py

```diff
@@ -242,6 +247,10 @@ def connection_made(self, transport):
         self.transport = transport
         Log.debug("Client connected to modbus server")
         self._connected = True
+
+        if self.params.after_connect_delay:
+            time.sleep(self.params.after_connect_delay / 1000)
```

Comment:
This is an async corutine, you cannot use time.sleep here, that blocks the whole loop.

Reply:
`connection_made` is sadly not an async function, even while it is part of the `asyncio.Protocol` interface.

Reply:
I did not write it is an async method, because it is not, but it is a coroutine (called from create_protocol, which are async), and thus calling time.sleep blocks the whole asyncio loop.

Coroutines are meant to do something fast, but not make big work, IO or other items that can block the rest of the system.